packages/netlify-cms-core/src/actions/config.js

```diff
@@ -245,9 +272,11 @@ export function applyDefaults(config) {
                   'fields',
                   traverseFields(file.get('fields'), setDefaultPublicFolder),
                 );
+                file = setI18nDefaults(i18n, file);
```

Comment:
files should inherit collection level configuration. I think calling `setI18nDefaults` this way will inherit top level configuration.

This will require re-using some of the logic from here:
https://github.com/netlify/netlify-cms/blob/519cb2d4c2db729d2643c9116f93656b6a9dba23/packages/netlify-cms-core/src/actions/config.js#L74